### PR TITLE
Updates feb 28 2022

### PR DESCRIPTION
**Pull request by rbeucher**

### CODE REVIEW

File: Jupyterbook/Notebooks/Ex_NavierStokesFlowAroundSphere.py [hunk 1]
```diff
@@ -135,24 +135,47 @@
...
 
 
 # +
 # Create NS object
 
-navier_stokes = NavierStokes(pipemesh, 
+navier_stokes = NavierStokesSwarm(pipemesh, 
                 velocityField=v_soln, 
                 pressureField=p_soln, 
+                velocityStar_fn=v_star,
```
Comment (rbeucher) [inline Jupyterbook/Notebooks/Ex_NavierStokesFlowAroundSphere.py]: @lmoresi you are passing a swarmVar here. Isn't it supposed to be an `fn` object?

Comment (lmoresi) [reply Jupyterbook/Notebooks/Ex_NavierStokesFlowAroundSphere.py]: Yes. That's what I have in my notebooks.

Comment (lmoresi) [reply Jupyterbook/Notebooks/Ex_NavierStokesFlowAroundSphere.py]: I have no idea how that managed to slip through. Let's add checks for the types !

Comment (rbeucher) [reply Jupyterbook/Notebooks/Ex_NavierStokesFlowAroundSphere.py]: Is it Optional that v_star?

**Comment by rbeucher** [comment]
OK @lmoresi . Doing a clean install from your branch I can't load Underworld.
Looks like there is some circular imports... 
@julesghub ?

File: Jupyterbook/Notebooks/Ex_NavierStokesFlowAroundSphere.py (hunk 1 above)
Comment (lmoresi) [reply Jupyterbook/Notebooks/Ex_NavierStokesFlowAroundSphere.py]: No, it should be required. Though we have mixed things up a bit - we also allow some required parameters to be specified after the solver has been instantiated. This is something John did quite a bit and I think it is to allow the pattern we discussed in uw2 where we establish a solver framework first and then fill in the pieces.

**Comment by lmoresi** [comment]
> OK @lmoresi . Doing a clean install from your branch I can't load Underworld.
> Looks like there is some circular imports...
> @julesghub ?

I had this happening when I included the argument type for the v_star_fn argument - that's why I had the type commented out in the __init__ function

**Pull request closed without merging**